eng/Packages.Data.props

```diff
@@ -245,7 +245,7 @@
     <PackageReference Update="Newtonsoft.Json" Version="10.0.3" />
     <PackageReference Update="NSubstitute" Version="3.1.0" />
     <PackageReference Update="NUnit" Version="3.13.2" />
-    <PackageReference Update="NUnit3TestAdapter" Version="4.2.1" />
+    <PackageReference Update="NUnit3TestAdapter" Version="4.3.0-alpha-net7.4" />
```

Comment:
In general we try to avoid prerelease packages because of stability issues. We should make a team decision before taking this update. /cc @pallavit @jsquire

Reply:
Certainly, for production, but this is test-only. Without it, VS Tests against .NET Framework don't run. Seems like a pretty serious gap.

Reply:
They don't run on a system with a preview version of .NET 7? That seems like a pretty big back compat bug on the VS side. Do you know if this is temporary?

Reply:
This alpha version was shipped 3 months ago.. Perhaps we should ask the NUnit folks when they plan on releasing a stable version.

Reply:
Perhaps they are waiting for the stable .NET 7..

Reply:
I'm good with this based on the need and scope of test-only.